hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataResolvers.java

```diff
@@ -57,7 +54,9 @@ public class KvMetadataResolvers {
 
     // A string of characters (excluding a `.`), optionally prefixed with "__key." or "this."
     private static final Pattern EXT_NAME_PATTERN = Pattern.compile("((" + KEY + "|" + VALUE + ")\\.)?[^.]+");
-    private static final Set<TypeKind> NESTED_FIELDS_SUPPORTED_FORMATS = ImmutableSet.of(JAVA, PORTABLE, COMPACT);
+    private static final Set<String> NESTED_FIELDS_SUPPORTED_FORMATS = new HashSet<>(Set.of(
```

Comment:
Key or value format might be omitted by user, in which case, the format is null. The immutable set generated by `Set.of()` throws exception if null is passed to `Set.contains()`, whereas `HashSet` allows checking the existence of null.

Reply:
if `null` format is expected it would be readable to include it explicitly in `if` condition instead of relying on the fact that `null` is not present in `NESTED_FIELDS_SUPPORTED_FORMATS`.

Reply:
407f4405

Reply:
actually the `format` is mandatory, see `KvMetadataResolvers.findMetadataResolver` but that check happens after nested fields enrichment. Is that really the most logical order of invocations?

Reply:
There is `KvMetadataNullResolver`, whose supported format is null. `KvMetadataResolver.resolveAndValidateFields()` needs to validate nested types recursively, and to be able to do this, it requires custom types to be resolved already.

Reply:
9a0e8973